src/rez/config.py

```diff
@@ -23,11 +23,44 @@
 import os
 import re
 import copy
+import warnings
 
 
 basestring = six.string_types[0]
 
 
+def _warn_deprecated_settings(data, filepath):
+    for key in data:
+        if key in _deprecated_settings:
+            message = (
+                "rez config setting named {0!r} is "
+                "deprecated and will be removed in 3.0.0."
+            ).format(key)
+
+            if os.getenv("REZ_LOG_ALL_DEPRECATION_WARNINGS"):
+                with warnings.catch_warnings():
```

Comment:
I don't like this. Why not just print to stderr? We could use `warnings.formatwarning` and print that to stderr.

We can also refactor this into two function. One that takes a message argument and another that takes no messaage. This will allow to centralize the warning logic in rez.deprecations (even for the config file).

Reply:
Or maybe just implement our own deprecation warning class?